### ISSUE TITLE
Provide an option for outputting JSON status objects similar to /teamcity

### ISSUE THREAD

**Issue by hach-que** [xunit/xunit#287]
The "-teamcity" option outputs the status of running tests as it occurs.  It'd be nice to have an option that outputs JSON objects separated by newlines as the status of running tests change, as this would make it easier to integrate progress indicators into other tools (because most languages can parse JSON).

Basically the output might look something like:

```
{ "type": "testSuiteStarted", "name": "Test collection for ...", "flowId": "..." }
{ "type": "testStarted", "name": "Test collection for ...", "flowId": "..." }
{ "type": "testFinished", "name": "Test collection for ...", "flowId": "...", "duration": 123 }
{ "type": "testFailed", "name": "Test collection for ...", "flowId": "...", "details": "..." }
```

**Comment by odedw** [comment]
I'd like to take a look at this.

**Comment by bradwilson** [comment]
Implemented with #647.
